src/animation/AnimationMixer.js

```diff
@@ -555,6 +555,13 @@ AnimationMixer.prototype = Object.assign( Object.create( EventDispatcher.prototy
 		// clip must be known when specified via string
 		if ( clipObject === null ) return null;
 
+		if ( needsValidateAndOptimize !== false ) {
+
+			clipObject.validate();
+			clipObject.optimize();
```

Comment:
From your scenario, I came to think of calling `.optimize()` only if tracks are valid. Users may want fix tracks if they're invalid but tracks data will be changed if we call `.optimize()`. What do you think?

Reply:
Done. ✅